### PR TITLE
feat(core): Add support for SQLite connection pooling

### PR DESCRIPTION
**Pull request by netroy**
This PR adds support for connection pooling with sqlite (disabled by default).

Related PR https://github.com/n8n-io/typeorm/pull/9

## Review / Merge checklist
- [x] PR title and summary are descriptive
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included

### CODE REVIEW

File: .github/workflows/ci-postgres-mysql.yml [hunk 1]
```diff
@@ -34,6 +34,33 @@ jobs:
           path: ./packages/**/dist
           key: ${{ github.sha }}:db-tests
 
+  sqlite-pooled:
```
Comment (tomi) [inline .github/workflows/ci-postgres-mysql.yml:37]: The file name no longer matches all the contents, but I think it's fine

File: packages/cli/src/databases/config.ts [hunk 2]
```diff
@@ -47,16 +48,23 @@ export const getOptionOverrides = (dbType: 'postgresdb' | 'mysqldb') => ({
...
+			config.getEnv('database.sqlite.database'),
+		),
+		enableWAL: config.getEnv('database.sqlite.enableWAL'),
+		migrations: sqliteMigrations,
+	};
+	if (poolSize > 0) {
+		return { type: 'sqlite-pooled', poolSize, ...commonOptions };
+	} else {
+		return { type: 'sqlite', ...commonOptions };
+	}
```
Comment (tomi) [inline packages/cli/src/databases/config.ts:66]: I was thinking we could have done this condition within the library, in the `DriverFactory`, but no objections to doing it here either

**Review by tomi — DISMISSED** [review]
👍 only couple small comments

File: packages/cli/src/config/schema.ts [hunk 3]
```diff
@@ -173,9 +173,15 @@ export const schema = {
 			enableWAL: {
 				doc: 'Enable SQLite WAL mode',
 				format: Boolean,
-				default: false,
+				default: true,
```
Comment (despairblue) [inline packages/cli/src/config/schema.ts:176]: Should this be marked as a breaking change, considering that WAL has some downsides, e.g. the file system has to support it, it's a bit slower and backup solutions need to be aware of additional files: 
https://www.sqlite.org/wal.html

Comment (netroy) [reply packages/cli/src/config/schema.ts]: updated this to be backward-compatible. now WAL is always enabled when using pooling, but by default, the env variable is respected.

File: packages/cli/src/databases/config.ts (hunk 2 above)
Comment (netroy) [reply packages/cli/src/databases/config.ts]: I want it to do it that way eventually as well, but for now, I want to get this moving without making any changes to the default sqlite driver. Once pooling seems mature enough, maybe we can just delete the old driver.

File: .github/workflows/ci-postgres-mysql.yml (hunk 1 above)
Comment (netroy) [reply .github/workflows/ci-postgres-mysql.yml:37]: didn't want to rename the file, and create a new entry a new workflow in the actions.

**Review by tomi — DISMISSED** [review]

**Review by tomi — DISMISSED** [review]
🚀

**Review by tomi — DISMISSED** [review]

**Review by tomi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.34.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.34.0)

**Pull request merged**